dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/DefaultTrackerObjectsDeletionService.java

```diff
@@ -113,7 +113,11 @@ public TrackerTypeReport deleteEnrollments(List<String> enrollments) throws NotF
       TrackedEntity te = enrollment.getTrackedEntity();
       te.setLastUpdatedByUserInfo(userInfoSnapshot);
 
-      apiEnrollmentService.deleteEnrollment(enrollment);
+      enrollment.setStatus(EnrollmentStatus.CANCELLED);
```

Comment:
Is something similar needed for events which also has a status?

Reply:
The PR to refactor the events deletion is this one: https://github.com/dhis2/dhis2-core/pull/18010/files
Doesn't look like we were using the status before either, I don't know why though.